posts/2021-04-01-open-liberty-devfile-stack-minikube.adoc

```diff
@@ -0,0 +1,127 @@
+
+== OpenShift Do (odo)
+
+Odo is a simple CLI tool that allows you to create stack projects and interact directly with your Kubernetes cluster (e.g. Minikube). Once you link:https://odo.dev/docs/installing-odo/[install odo], you can start creating Open Liberty stack projects and developing right in Kubernetes.
+
+== Try it out
+
+After link:https://minikube.sigs.k8s.io/docs/start/[installing Minikube], start your local cluster:
+
+[source,sh]
+----
+$ minikube start
+----
+
+MacOS users may need to start Minikube with a driver other than "docker" due to a known issue with the docker driver and the ingress addon. For example: `--driver=virtualbox`
+
+Next, enable the Minikube ingress addon:
+
+[source,sh]
+----
+$ minikube addons enable ingress
+----
+
+This will allow you to create a URL for your app using an ingress domain.
+
+Start the Minikube dashboard (Optional):
+
+[source,sh]
+----
+$ minikube dashboard
+----
+
+Minikube will open a k8s dashboard in your browser. While not necessary, it allows you to view and interact with your cluster and your development pod. 
+
+
+With Minikube set up, you are now ready to create an Open Liberty stack project. You can either initialize an existing application or use the link:https://github.com/OpenLiberty/application-stack-starters[stack starter] to create a new app. For this blog, we are going to use the link:https://github.com/OpenLiberty/application-stack-intro/tree/blog-demo[application-stack-intro] app. This application is a basic JAX-RS sample that uses Maven, Open Liberty, and MicroProfile.
+
+[source,sh]
+----
+$ git clone -b blog-demo git@github.com:OpenLiberty/application-stack-intro.git && \
+  cd application-stack-intro
+----
+
+Initialize the app as a "java-openliberty" stack project:
+
+[source,sh]
+----
+$ odo create java-openliberty my-ol-comp
+----
+
+You will now see the addition of a `devfile.yaml` file that was downloaded from the default devfile registry. This is the brains behind the Open Liberty stack and defines the development environment for your app. 
+
+Create a URL for the application using an Ingress domain:
+
+For convenience, you can use the link:https://nip.io/[nip.io] service to create mappings of your ingress controller to an external IP. Otherwise, you will need to add the IP to your `etc/hosts` file.
+
+[source,sh]
+----
+$ odo url create --host $(minikube ip).nip.io
+----
+
+Push your app to your local Minikube cluster:
+
+[source,sh]
+----
+$ odo push
+----
+
+Check the output to make sure that your application is up and ready for use:
+
+[source,sh]
+----
+$ odo log
```

Comment:
Maybe give a hint as to what to look for for readiness?  It can be a long startup.. especially if they tweak it.

Reply:
I added the "smarter planet" message which should be a good indication.